data_structures/queue/queue_on_two_stacks.py

```diff
@@ -0,0 +1,138 @@
+"""Queue implementation using two stacks"""
+
+from collections.abc import Iterable
+from typing import Generic, TypeVar
+
+_T = TypeVar("_T")
+
+
+class Queue(Generic[_T]):
+    def __init__(self, iterable: Iterable[_T] | None = None) -> None:
+        """
+        >>> queue1 = Queue()
+        >>> str(queue1)
+        '<[]>'
+        >>> queue2 = Queue([10, 20, 30])
+        >>> str(queue2)
+        '<[10, 20, 30]>'
+        >>> queue3 = Queue((i**2 for i in range(1, 4)))
+        >>> str(queue3)
+        '<[1, 4, 9]>'
+        """
+
+        self._stack1: list[_T] = [] if iterable is None else list(iterable)
+        self._stack2: list[_T] = []
+
+    def put(self, item: _T) -> None:
+        """
+        Put `item` into the Queue
+
+        >>> queue = Queue()
+        >>> queue.put(10)
+        >>> queue.put(20)
+        >>> len(queue) == 2
+        True
+        >>> str(queue)
+        '<[10, 20]>'
+        """
+
+        self._stack1.append(item)
+
+    def get(self) -> _T:
+        """
+        Get `item` from the Queue
+
+        >>> queue = Queue()
+        >>> for i in (10, 20, 30):
+        ...     queue.put(i)
+        >>> len(queue)
+        3
+        >>> queue.get()
+        10
+        >>> queue.get()
+        20
+        >>> len(queue) == 1
+        True
+        >>> queue.get()
+        30
+        >>> queue.get()
+        Traceback (most recent call last):
+            ...
+        IndexError: Queue is empty
+        """
+
+        # To reduce number of attribute look-ups in `while` loop.
+        stack1_pop = self._stack1.pop
+        stack2_append = self._stack2.append
+
+        if not self._stack2:
+            while self._stack1:
+                stack2_append(stack1_pop())
+
+        if not self._stack2:
+            raise IndexError("Queue is empty")
+        return self._stack2.pop()
+
+    def size(self) -> int:
+        """
+        Returns the length of the Queue
+
+        >>> queue = Queue()
+        >>> queue.size()
+        0
+        >>> queue.put(10)
+        >>> queue.put(20)
+        >>> queue.size()
+        2
+        >>> queue.get()
+        10
+        >>> queue.size() == 1
+        True
+        """
+
+        return len(self)
+
+    def __str__(self) -> str:
+        """
+        >>> queue = Queue()
+        >>> str(queue)
+        '<[]>'
+        >>> queue.put(10)
+        >>> str(queue)
+        '<[10]>'
+        >>> queue.put(20)
+        >>> str(queue)
+        '<[10, 20]>'
```

Comment:
https://docs.python.org/3/reference/datamodel.html#object.\_\_repr\_\_

The output should be the same as the command to create a new instance with these contents.

```suggestion
        'Queue(10, 20)'
```

Reply:
@cclauss  Yes totally agree. It's True for `__repr__`. I'm gonna do that next and change the tests.  One more question `Queue` doesn't accept `*args` but an iterable so the representation can't be `Queue(10, 20, 30)`. Is it okay if I wrap them in bracket? like `Queue([10, 20, 30])`.